### DISCUSSION TITLE
Fully Customizable Home Page for Modifying Cards and Counts

### DISCUSSION THREAD

**Discussion by JFman98** [netbox-community/netbox#7800] [Ideas]
I am looking to be able to customize the layout of the home page. For example, our organization does not use Power and Circuits. I would like to hide these cards from the home page. Our organization also uses services and would like to see this under the IPAM card. We would also like to remove tenants from the Organization card. I propose creating a fully customizable home page to allow for quicker clicking on items rather than fishing though the side nav bar. This would also allow for quick counts of items that an organization cares about most.

I do not know how to implement this as I am just a systems admin, but if there is anyone else who would like to see this become a feature, I would love to see some technical input.

Thanks,
Justin

**Comment by JFman98** [comment]
I edited the source of our active site to show an example of what we are looking for. Every organization might have different use cases, so making it editable in the preferences would be nice.

![image](https://user-images.githubusercontent.com/14074253/141172601-76a43661-5938-47ba-b017-dc6bbe3992b8.png)

**Comment by candlerb** [comment]
There was a separate feature request for disabling entire sections of Netbox functionality entirely - #426 - together with related ones like #3108.

I think your idea of customizing front page is better, especially if this could be done for different groups of users, as the feature could still exist but only usable by those who need it.

I wonder if it could be linked to permissions: if a user doesn't have permissions to view circuits (or whatever), then instead of displaying the objects with a lock icon beside, don't display them at all.  This could also apply to menus in the side bar.

**Reply by jeremystretch** [reply]
> I wonder if it could be linked to permissions: if a user doesn't have permissions to view circuits (or whatever), then instead of displaying the objects with a lock icon beside, don't display them at all. This could also apply to menus in the side bar.

We intentionally don't do this because it leads to user confusion. They'll read about some feature in the documentation, wonder why they don't see it, and ask why it's missing (probably here or on Slack). IMO it's better to clearly show that the feature exists but that they don't have access to it.

**Reply by candlerb** [reply]
Maybe, but I'd argue that someone who's administering Netbox will be able to login as a "superuser" and see everything, so there won't be any problem.

Anyone who's not administering Netbox is just an end-user, and should be talking to their local administrator first if there's a problem.

**Reply by esoubihe** [reply]
I agree with @candlerb. With the growing number of features supported by Netbox, admins might want to streamline views for their users based on what's actually being used. The user profile page could contain a list of models the user doesn't have access to if you feel that would contribute to less confusion, but my opinion is that RBAC should be reflected throughout the app.
